datajunction-query/djqs/engine.py

```diff
@@ -66,9 +66,30 @@ def get_columns_from_description(
     return columns
 
 
+def create_trino_engine(engine: Engine, headers: Optional[Dict[str, str]]) -> "Engine":
+    """
+    Create a SQLAlchemy engine for Trino.
+    """
+    uri = engine.uri
+    user = engine.extra_params["user"]
+    password = engine.extra_params["password"]
+    scheme, rest = uri.split("://")
```

Comment:
Can you use `urlparse` here instead? https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse 

Then extracting the `scheme`, `host`, and `port` from the uri can be much simplified: 
```
from urllib.parse import urlparse
parsed = urlparse(engine.uri)
print(parsed.scheme, parsed.port, parsed.hostname)
```